sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -0,0 +1,595 @@
+        {
+          throw new UnsupportedOperationException("Do not use iterator(), it cannot be closed.");
+        }
+
+        @Override
+        public Enumerator<Object[]> enumerator()
+        {
+          return new Enumerator<Object[]>()
+          {
+            @Override
+            public Object[] current()
+            {
+              TaskStatusPlus task = it.next();
+              return new Object[]{task.getId(), task.getType(),
+                                  task.getDataSource(),
+                                  task.getCreatedTime(),
+                                  task.getQueueInsertionTime(),
+                                  task.getState(),
+                                  task.getRunnerTaskState(),
+                                  task.getDuration(),
+                                  task.getLocation() != null
+                                  ? task.getLocation().getHost() + ":" + (task.getLocation().getTlsPort()
+                                                                          == -1
+                                                                          ? task.getLocation()
+                                                                                .getPort()
+                                                                          : task.getLocation().getTlsPort())
+                                  : null,
+                                  task.getLocation().getHost(),
+                                  task.getLocation().getPort(),
+                                  task.getLocation().getTlsPort(),
+                                  task.getErrorMsg()};
+            }
+
+            @Override
+            public boolean moveNext()
+            {
+              return it.hasNext();
+            }
+
+            @Override
+            public void reset()
+            {
+
+            }
+
+            @Override
+            public void close()
+            {
+              try {
+                it.close();
+              }
+              catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+          };
+        }
+      }
+
+      return new TasksEnumerable(getTasks(druidLeaderClient, jsonMapper, responseHandler));
+    }
+  }
+
+  //Note that overlord must be up to get tasks
+  private static JsonParserIterator<TaskStatusPlus> getTasks(
+      DruidLeaderClient indexingServiceClient,
+      ObjectMapper jsonMapper,
+      BytesAccumulatingResponseHandler responseHandler
+  )
+  {
+
+    Request request;
+    try {
+      request = indexingServiceClient.makeRequest(
+          HttpMethod.GET,
+          StringUtils.format("/druid/indexer/v1/tasks")
+      );
+    }
+    catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    ListenableFuture<InputStream> future = indexingServiceClient.goStream(
+        request,
+        responseHandler
+    );
+    try {
+      future.get();
```

Comment:
Same here. Is this intentional?

Reply:
yes

Reply:
Please check my comment [here](https://github.com/apache/incubator-druid/pull/6094#discussion_r223201970).

Reply:
modified this as well